server/api_server/nettools_endpoint.py

```
@@ -2,12 +2,37 @@
+# Resolve speedtest-cli path once at module load and validate it.
+# We do this once to avoid repeated PATH lookups and to fail fast when
+# the binary isn't available or executable.
+SPEEDTEST_CLI_PATH = None
+
+def _get_speedtest_cli_path():
+    """Resolve and validate the speedtest-cli executable path."""
+    path = shutil.which("speedtest-cli")
+    if path is None:
+        raise RuntimeError(
+            "speedtest-cli not found in PATH. Please install it: pip install speedtest-cli"
+        )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)